astropy/io/fits/hdu/image.py

```diff
@@ -638,7 +646,49 @@ def _writedata_internal(self, fileobj):
 
             size += output.size * output.itemsize
 
-        return size
+            return size
+
+    def _writeinternal_dask(self, fileobj):
+
+        if sys.byteorder == 'little':
+            swap_types = ('<', '=')
+        else:
+            swap_types = ('<',)
+        # deal with unsigned integer 16, 32 and 64 data
+        if _is_pseudo_unsigned(self.data.dtype):
+            raise NotImplementedError("This dtype isn't currently supported with dask.")
+        else:
+            output = self.data
+            byteorder = output.dtype.str[0]
+            should_swap = (byteorder in swap_types)
+
+        if should_swap:
+            from dask.utils import M
+            # NOTE: the byteswap flag needs to be False otherwise the array is
```

Comment:
Maybe this is clearer?
```suggestion
            # NOTE: the inplace flag to byteswap needs to be False otherwise the array is
```